Application/Api/Document/Custom/Limbach/Repository/SchoolContract.php

```diff
@@ -0,0 +1,422 @@
+        $this->FieldValue['LastNameCustody2'] = (isset($DataPost['LastNameCustody2']) && $DataPost['LastNameCustody2'] != '' ? $DataPost['LastNameCustody2'] : '&nbsp;');
+//        $this->FieldValue['AddressDistrict2'] = (isset($DataPost['AddressDistrict2']) && $DataPost['AddressDistrict2'] != '' ? $DataPost['AddressDistrict2'] : '&nbsp;');
+        $this->FieldValue['AddressStreet2'] = (isset($DataPost['AddressStreet2']) && $DataPost['AddressStreet2'] != '' ? $DataPost['AddressStreet2'] : '&nbsp;');
+        $this->FieldValue['AddressPLZ2'] = (isset($DataPost['AddressPLZ2']) && $DataPost['AddressPLZ2'] != '' ? $DataPost['AddressPLZ2'] : '&nbsp;');
+        $this->FieldValue['AddressCity2'] = (isset($DataPost['AddressCity2']) && $DataPost['AddressCity2'] != '' ? $DataPost['AddressCity2'] : '&nbsp;');
+
+        // Compare Custody Address
+        $this->FieldValue['IsCompare'] = false;
+        if($this->FieldValue['AddressStreet1'] == $this->FieldValue['AddressStreet2']
+        && $this->FieldValue['AddressPLZ1'] ==  $this->FieldValue['AddressPLZ2']
+        && $this->FieldValue['AddressCity1'] ==  $this->FieldValue['AddressCity2']){
+            $this->FieldValue['IsCompare'] = true;
+        }
+        // one Custody
+        if($this->FieldValue['LastNameCustody2'] == '&nbsp;'){
+            $this->FieldValue['IsCompare'] = true;
+        }
+
+        // Custody
+        $this->FieldValue['CustodyString'] = '';
+        if($this->FieldValue['LastNameCustody1'] == $this->FieldValue['LastNameCustody2'] && $this->FieldValue['LastNameCustody2'] != '&nbsp;'){
+            $this->FieldValue['CustodyString'] =
+                $this->FieldValue['LastNameCustody1'].' '.$this->FieldValue['FirstNameCustody1'].' und '.
+                $this->FieldValue['FirstNameCustody2'];
+        } else {
+            $this->FieldValue['CustodyString'] =
+                $this->FieldValue['LastNameCustody1'].' '.$this->FieldValue['FirstNameCustody1']
+                .($this->FieldValue['FirstNameCustody2'] != '&nbsp;'
+                    ? ' und '.$this->FieldValue['LastNameCustody2'].' '.$this->FieldValue['FirstNameCustody2']
+                    : '');
+        }
+
+        // Gender
+        $this->FieldValue['Male'] = 'false';
+        $this->FieldValue['Female'] = 'false';
+        if (isset($this->FieldValue['Gender']) && $this->FieldValue['Gender'] == 'Männlich') {
+            $this->FieldValue['Male'] = 'true';
+        } elseif (isset($this->FieldValue['Gender']) && $this->FieldValue['Gender'] == 'Weiblich') {
+            $this->FieldValue['Female'] = 'true';
+        }
+
+        return $this;
+    }
+
+    /**
+     * @return string
+     */
+    public function getName()
+    {
+
+        return 'Schulvertrag';
+    }
+
+    /**
+     * @param array $pageList
+     *
+     * @return Frame
+     */
+    public function buildDocument($pageList = array())
```

Comment:
Codacy found an issue: [The method buildDocument() has 207 lines of code. Current threshold is set to 100. Avoid really long methods.](https://app.codacy.com/manual/gerdchristian-kunze/SPHERE-Framework/pullRequest?prid=5757982)